models/articles.model.js

```diff
@@ -2,13 +2,32 @@ const db = require('../db/connection');
 
 exports.fetchArticleById = (article_id) => {
   return db.query(
-      'SELECT author, title, article_id, body, topic, created_at, votes, article_img_url FROM articles WHERE article_id = $1',
-      [article_id]
-    )
-    .then((result) => {
-    
-      return result.rows[0]
-    })
+    `SELECT 
+        articles.author, 
+        articles.title, 
+        articles.article_id, 
+        articles.body, 
+        articles.topic, 
+        articles.created_at, 
+        articles.votes, 
+        articles.article_img_url,
+        COUNT(comments.article_id)::INT AS comment_count
+     FROM articles
+     LEFT JOIN comments 
+     ON articles.article_id = comments.article_id
+     WHERE articles.article_id = $1
+     GROUP BY articles.article_id`, 
+    [article_id]
+  )
+  .then((result) => {
+    if (result.rows.length === 0) {
+      return null;
```

Comment:
You should be returning a promise.reject here with a custom error - you might be handling it another way but I would have a look at refactoring at some point if you can :)